spec/models/favorite_spec.rb

```diff
@@ -48,4 +48,51 @@
       expect(favorite.favorite_pets).to eq([pet_1])
     end
   end
+
+  describe "#favorite_already?" do
+    it "checks if pet was added to favorites already" do
+      shelter_1 = Shelter.create(name: "Denver Animal Shelter",
+                                  address: "3301 Navajo Street",
+                                  city: "Denver",
+                                  state: "CO",
+                                  zip: 80021)
+
+      pet_1 =  Pet.create(name: "Winnie",
+                          approximate_age: 3,
+                          sex: "Female",
+                          image: "https://imgur.com/r/puppies/cYqJGNo",
+                          adoption_status: "Available",
+                          shelter_id: shelter_1.id)
+
+      favorite = Favorite.new(pet_1.id => 1)
+
+      favorite.add_pet(pet_1)
+
+      expect(favorite.favorite_already?(pet_1.id)).to eq(1)
+      expect(favorite.favorite_already?(pet_1.id)).to_not eq(2)
+    end
+  end
+
+  describe "#remove_pet" do
+    it "takes pet out of favorites" do
+      shelter_1 = Shelter.create(name: "Denver Animal Shelter",
+                                  address: "3301 Navajo Street",
+                                  city: "Denver",
+                                  state: "CO",
+                                  zip: 80021)
+
+      pet_1 =  Pet.create(name: "Winnie",
+                          approximate_age: 3,
+                          sex: "Female",
+                          image: "https://imgur.com/r/puppies/cYqJGNo",
+                          adoption_status: "Available",
+                          shelter_id: shelter_1.id)
+
+      favorite = Favorite.new(pet_1.id => 1)
+
+      favorite.remove_pet(pet_1)
+
+      expect(favorite.count_of(pet_1.id)).to eq(0)
```

Comment:
Given your comment about having issues where deleting favorites deleted the pet, we should add a capybara line expecting that the pet will still exist.